tests/kernel/mem_pool/sys_mem_pool/src/main.c

```diff
@@ -95,6 +95,7 @@ void test_sys_mem_pool_alloc_align4(void)
  */
 void test_sys_mem_pool_min_block_size(void)
 {
+#ifndef CONFIG_SYS_HEAP_MEMPOOL_COMPAT
```

Comment:
There is a ztest_test_skip() utility that would probably be better applied here than block preprocessor conditionals.  Also we should include a comment explaining why this particular test case isn't applicable, e.g.:

```c
if (IS_ENABLED(CONFIG_SYS_HEAP_MEMPOOL_COMPAT)) {
    /* Doesn't work with compat API because of {alignment/whatever...} */
    ztest_test_skip();
    return;
}